xc/xenops_server_xen.ml

```diff
@@ -2928,6 +2958,9 @@ module Actions = struct
 							"Failed to deregister RRD plugin: caught %s"
 							(Printexc.to_string e)
 				end
+			| "local" :: "domain" :: domid :: "qemu-pid-signal" :: [] ->
+				let value = Device.Qemu.signal ~xs (int_of_string domid) in
```

Comment:
I'm also using this function in device.waitpid_async. A second reason I've added it isi because the DaemonMgmt module had a "pid" function to extract the value of the pid key, so it seemed reasonable to add a "signal" function to extract the value of the signal key. I will change both places to use xs.Xs.read.